### PR TITLE
test: properly clean up cilium components in ginkgo contexts

### PR DESCRIPTION
**Pull request by fristonio**
See individual commit messages for details.

### CODE REVIEW

**Comment by fristonio** [comment]
retest-gke

**Comment by fristonio** [comment]
retest-gke

**Comment by fristonio** [comment]
retest-gke

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -1465,6 +1465,14 @@ func (kub *Kubectl) Delete(filePath string) *CmdRes {
 		fmt.Sprintf("%s delete -f  %s", KubectlCmd, filePath))
 }
 
+// DeleteAndWait deletes the Kubernetes manifest at path filePath and wait
+// for the associated resources to be gone.
+func (kub *Kubectl) DeleteAndWait(filePath string) *CmdRes {
+	kub.Logger().Debugf("waiting for resources in %q to be deleted", filePath)
+	return kub.ExecShort(
```
Comment (nebril) [inline test/helpers/kubectl.go]: I would use at least `ExecMiddle` here, 10 seconds will probably cause flakes here.

File: test/k8sT/Chaos.go [hunk 2]
```diff
@@ -153,9 +155,7 @@ var _ = Describe("K8sChaosTest", func() {
 			By("Checking connectivity after uninstalling Cilium")
 			connectivityTest()
 
-			By("Install cilium pods")
-
-			ciliumFilename := helpers.TimestampFilename("cilium.yaml")
+			By("ReInstall cilium DaemonSet")
```
Comment (nebril) [inline test/k8sT/Chaos.go]: nitpick: `s/ReInstall/Reinstall/`

**Review by nebril — CHANGES_REQUESTED** [review]
Looks good overall, great change! I would really like for delete to have bigger timeout than 10 seconds, aside from that let's ship it!

File: test/helpers/kubectl.go (hunk 1 above)
Comment (fristonio) [reply test/helpers/kubectl.go]: Yeah, that makes sense. Updated the command.

**Comment by fristonio** [comment]
retest-gke

**Comment by fristonio** [comment]
retest-gke

**Comment by fristonio** [comment]
retest-gke

**Comment by fristonio** [comment]
retest-4.19

**Comment by fristonio** [comment]
retest-4.19

**Comment by fristonio** [comment]
retest-gke

**Comment by fristonio** [comment]
retest-4.9

**Comment by fristonio** [comment]
retest-runtime

**Comment by fristonio** [comment]
retest-net-next

**Review by nebril — APPROVED** [review]
Let's ship it!

**Pull request merged**